src/components/Feedback.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState, useRef } from "@wordpress/element";
+import { __ } from "@wordpress/i18n";
+//
+import { InteractionAPIs } from "../utils";
+
+const Feedback = ({ postId }) => {
+  const [status, setStatus] = useState("");
+  const yesButtonRef = useRef(null);
+  const noButtonRef = useRef(null);
+
+  const postFeedback = async () => {
+    if (status === "helpful" || status === "notHelpful") {
+      InteractionAPIs.postFeedback(postId, status);
+    }
+  };
+
+  useEffect(() => {
+    setStatus("");
+    noButtonRef.current.className = "feedback-button no";
+    yesButtonRef.current.className = "feedback-button yes";
+  }, [postId]);
+
+  useEffect(() => {
```

Comment:
Would using a state instead of ref have any significant impact on performance or something ?

Reply:
It's more idiomatic React to use state here instead of refs. Using state also means we don't need to perform cleanups using useEffect on the refs.

For a small component like this, there's no discussion to be had on performance.

Reply:
This class name does not really contribute to any logical state change in the page thus, we do not need to re-render the entire component just to add a class, secondly, the useRef has manipulating DOM listed as one of the primary use-case: https://react.dev/reference/react/useRef#manipulating-the-dom-with-a-ref

 

So I guess both in terms of performance and what React suggests, useRef looks like a good option to me.